src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -249,26 +265,145 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
+                        submissionFormHtml, "", null, null));
+            }
             responseIndx++;
         }
 
         return responses;
     }
 
+    private List<String> getRecipientList(List<String> responseSubmittedRecipient, Map<String, String> sortedRecipientList) {
+        List<String> recipientList = new ArrayList<String>();
+        for (String recipient : sortedRecipientList.values()) {
+            if (!responseSubmittedRecipient.contains(recipient)) {
+                recipientList.add(recipient);
+            }
+        }
+        return recipientList;
+    }
+
+    private FeedbackResponseCommentRow getResponseCommentsForQuestion(
+            FeedbackQuestionAttributes questionAttributes,
+            List<FeedbackResponseCommentAttributes> frcList, String giverName,
+            String recipientName, Map<FeedbackParticipantType, Boolean> responseVisibilityMap, Map<String,
+            String> commentGiverEmailNameTable) {
+        ZoneId sessionTimeZone = bundle.feedbackSession.getTimeZone();
+        if (isFeedbackSessionForInstructor) {
+            for (FeedbackResponseCommentAttributes frcAttributes : frcList) {
+                if (frcAttributes.giverEmail.equals(account.email)) {
+
+                    FeedbackResponseCommentRow frcRow = new FeedbackResponseCommentRow(frcAttributes,
+                            frcAttributes.giverEmail, giverName, recipientName,
+                            getResponseCommentVisibilityString(frcAttributes, questionAttributes),
+                            getResponseCommentGiverNameVisibilityString(frcAttributes, questionAttributes),
+                            responseVisibilityMap, commentGiverEmailNameTable, sessionTimeZone);
+                    setEditDeleteCommentOptionForUser(frcAttributes, frcRow);
+                    return frcRow;
+                }
+            }
+        } else {
+            for (FeedbackResponseCommentAttributes frcAttributes : frcList) {
+                if (frcAttributes.giverEmail.equals(student.email) || frcAttributes.giverEmail.equals(student.team)) {
+
+                    FeedbackResponseCommentRow frcRow = new FeedbackResponseCommentRow(frcAttributes,
+                            frcAttributes.giverEmail, giverName, recipientName,
+                            getResponseCommentVisibilityString(frcAttributes, questionAttributes),
+                            getResponseCommentGiverNameVisibilityString(frcAttributes, questionAttributes),
+                            responseVisibilityMap, commentGiverEmailNameTable, sessionTimeZone);
+                    setEditDeleteCommentOptionForUser(frcAttributes, frcRow);
+                    return frcRow;
+                }
+            }
+        }
+        return null;
+    }
+
+    private void setEditDeleteCommentOptionForUser(FeedbackResponseCommentAttributes frcAttributes,
+            FeedbackResponseCommentRow frcRow) {
+        if (isModeration) {
```

Comment:
can reduce the if else blocks here since the action is the same

Reply:
done